x-pack/plugins/maps/public/classes/sources/es_search_source/create_source_editor.js

```diff
@@ -64,12 +64,12 @@ export class CreateSourceEditor extends Component {
   }
 
   _onIndexPatternSelect = (indexPatternId) => {
-    this.setState(
-      {
-        indexPatternId,
-      },
-      this._loadIndexPattern(indexPatternId)
-    );
+    this.setState({ indexPatternId }, () => {
+      if (!indexPatternId) {
+        this._previewLayer();
+      }
+      this._loadIndexPattern(indexPatternId);
```

Comment:
Should we surround this line in an `else` block? If there's no `indexPatternId` we probably shouldn't be calling it

Reply:
I think it still makes sense to call loadIndexPattern since that function resets geofield as well. I can remove the changes to this file. They are fixing an unrelated bug that prevents `this.props.onSourceConfigChange` from getting called when index pattern is cleared. I think es_grid_source and es_pew_pew source have the same problems. I will move this into a separate PR

```
_loadIndexPattern = (indexPatternId) => {
    this.setState(
      {
        isLoadingIndexPattern: true,
        ...RESET_INDEX_PATTERN_STATE,
      },
      this._debouncedLoad.bind(null, indexPatternId)
    );
  };
```

Reply:
created https://github.com/elastic/kibana/issues/67618 to track